tests/python/test_tools_smoke.py

```diff
@@ -0,0 +1,371 @@
+        #      this was what we asked for using kill=True.
+        self.assertTrue((rc == 0 and allow_early) or rc == 124
+                        or (rc == 137 and kill), "rc was %d" % rc)
+
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def test_argdist(self):
+        self.run_with_duration("argdist.py -C 'p::SyS_open()' -n 1 -i 1")
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_bashreadline(self):
+        self.run_with_int("bashreadline.py")
+
+    def test_biolatency(self):
+        self.run_with_duration("biolatency.py 1 1")
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_biosnoop(self):
+        self.run_with_int("biosnoop.py")
+
+    def test_biotop(self):
+        self.run_with_duration("biotop.py 1 1")
+
+    def test_bitesize(self):
+        self.run_with_int("biotop.py")
+
+    def test_btrfsdist(self):
+        # Will attempt to do anything meaningful only when btrfs is installed.
+        self.run_with_duration("btrfsdist.py 1 1")
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_btrfsslower(self):
+        # Will attempt to do anything meaningful only when btrfs is installed.
+        self.run_with_int("btrfsslower.py", allow_early=True)
+
+    def test_cachestat(self):
+        self.run_with_duration("cachestat.py 1 1")
+
+    def test_cachetop(self):
+        # TODO cachetop doesn't like to run without a terminal, disabled
+        # for now.
+        # self.run_with_int("cachetop.py 1")
+        pass
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_capable(self):
+        self.run_with_int("capable.py")
+
+    def test_cpudist(self):
+        self.run_with_duration("cpudist.py 1 1")
+
+    @skipUnless(kernel_version_ge(4,9), "requires kernel >= 4.9")
+    def test_cpuunclaimed(self):
+        self.run_with_duration("cpuunclaimed.py 1 1")
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_dbslower(self):
+        # Deliberately left empty -- dbslower requires an instance of either
+        # MySQL or PostgreSQL to be running, or it fails to attach.
+        pass
+
+    @skipUnless(kernel_version_ge(4,3), "requires kernel >= 4.3")
+    def test_dbstat(self):
+        # Deliberately left empty -- dbstat requires an instance of either
+        # MySQL or PostgreSQL to be running, or it fails to attach.
+        pass
+
+    @skipUnless(kernel_version_ge(4,4), "requires kernel >= 4.4")
+    def test_dcsnoop(self):
+        self.run_with_int("dcsnoop.py")
+
+    def test_dcstat(self):
+        self.run_with_duration("dcstat.py 1 1")
+
+    @skipUnless(kernel_version_ge(4,6), "requires kernel >= 4.6")
+    def test_deadlock_detector(self):
+        # TODO This tool requires a massive BPF stack traces table allocation,
+        # which might fail the run or even trigger the oomkiller to kill some
```

Comment:
@4ast I made it huge originally in order to be able trace very large binaries with a large number of threads. The sizes in those maps should be made configurable.

Is there an easy way to pass arguments to the bpf C code from the user space python program, besides inlining the C program into the python program? Currently, the deadlock_detector bpf code exists as a separate C file.